### ISSUE TITLE
Consistent Input Microstutters

### ISSUE THREAD

**Issue by kvietcong** [ppy/osu#23448]
### Type

Performance

### Bug description

When moving the mouse around, it seems to stop accepting input every few seconds (video has the stutters captured). I've tried switching renderers and testing out single/multi threading. The task manager shows nothing weird during the micro stutters and game mode is enabled for windows.

This wasn't happening 3 releases ago (I just updated). Most other issues on stuttering seem to also include frame stuttering but it doesn't seem to be the case here.

Nvidia GTX 1660ti with 6GB VRAM, i7-9750H with 16GB RAM, SSD storage, Windows 11 22H2

The osu stable version seems to be fine.

### Screenshots or videos

This video is me moving my mouse back and forth and you can see consistent input micro stutters in fairly consistent intervals.

https://github.com/ppy/osu/assets/32186663/5d8cd492-26cf-4172-a3ce-cf68c3e4331b

It also happens in game and not just menus.

### Version

2023.419.0

### Logs

[network.log](https://github.com/ppy/osu/files/11429772/network.log)
[performance.log](https://github.com/ppy/osu/files/11429773/performance.log)
[runtime.log](https://github.com/ppy/osu/files/11429774/runtime.log)
[updater.log](https://github.com/ppy/osu/files/11429775/updater.log)
[database.log](https://github.com/ppy/osu/files/11429776/database.log)
[input.log](https://github.com/ppy/osu/files/11429777/input.log)

**Comment by peppy** [comment]
This generally happens when another app on your system is holding up the OS event queue. Check for any recent changes in applications you're running – try shutting absolutely everything down first and if that stops the stutters, start things back up until you find the cause.

**Comment by kvietcong** [comment]
I have already tried shutting down essentially everything which includes: Browsers, Discord, Overwolf, ShareX, and Vanguard (Riot's Anitcheat). Sadly it didn't alleviate the problem

**Comment by peppy** [comment]
Could be hardware or usb or something. Can't help much other than this sorry.

**Comment by artun42** [comment]
I've had the same issue, I solved it by setting my refresh rate to 2x of my display's (120hz display -> 240fps limit).

**Comment by kvietcong** [comment]
I have tested around with the various frame limits but that sadly didn't help either. It didn't make a difference to how long or how often the stutters happened to my perception

**Comment by kvietcong** [comment]
Tried uninstalling (wiped all lazer data) but that didn't help. Strange thing is that my tablet (pen tablet for input) works completely fine. No stutters whatsoever. Changing skins also doesn't help. I wonder if there was a change in mouse input within the last few months

**Comment by peppy** [comment]
I don't know how I can stress this further: something else on your system is causing the issue. That's precisely what the frame graph shows.

The fact it only happens for mouse confirms this more so: another app is handling global mouse events badly.

There may be a future where we can fix this our end by moving some more input handling code off the main event loop thread.

But for now you're going to have to figure it out at your end sorry.

The user above with the macOS issue is touching on a completely different issue which is known (and a different cause with different symptoms).

**Comment by kvietcong** [comment]
Thanks for the suggestion to look at global mouse events. I looked into the issue the past few days and it seems I have solved it, though with a few weird inconsistencies.

It seems the main issue was a Bluetooth keyboard (using [ZMK](https://zmk.dev/)) was interfering with my mouse's (Razer Orochi v2) 2.4ghz dongle (BT also uses 2.4ghz). The big inconsistency was that not everything had stuttering like in osu!lazer I tested it out on the following things:
- osu!lazer: Stutters
- osu!stable: No stutters
- Minecraft: Stutters
- Valorant: Stutters
- Windows desktop: No stutters
- Browser Games: No stutters

I have no idea what causes these different situations to have or have not stutters and all I can do now is think there is some type of interpolation/smoothing happening to mouse input for the places that don't stutter.

But disabling my keyboard's BT connection w/ my laptop fixed the issues. For people who also use ZMK, tuning the BT power output (turning in down) *doesn't* help.
